SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,430 @@
+{
+	const auto Strat = TestData->Strat;
+	const auto& TestActors = TestData->TestActors;
+
+	const bool WeShouldHaveAuthority
+		= Strat->WhoShouldHaveAuthority(*TestActors[ActorName]) == Strat->GetLocalVirtualWorkerId();
+	const bool DoWeActuallyHaveAuthority = Strat->ShouldHaveAuthority(*TestActors[ActorName]);
+
+	Test->TestEqual(
+		FString::Printf(TEXT("Expected: %b Actual: %b"), WeShouldHaveAuthority, DoWeActuallyHaveAuthority),
+		WeShouldHaveAuthority, DoWeActuallyHaveAuthority);
+	return true;
+}
+
+LAYEREDLBSTRATEGY_TEST(GIVEN_strat_is_not_ready_WHEN_local_virtual_worker_id_is_set_THEN_is_ready)
+{
+	AutomationOpenMap("/Engine/Maps/Entry");
+
+	TSharedPtr<TestData> Data = TSharedPtr<TestData>(new TestData);
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FCreateStrategy(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetDefaultLayer(Data, UGridBasedLBStrategy::StaticClass()));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(Data, {}));
+	ADD_LATENT_AUTOMATION_COMMAND(FCheckStratIsReady(Data, this, false));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategyLocalWorker(Data, 1));
+	ADD_LATENT_AUTOMATION_COMMAND(FCheckStratIsReady(Data, this, true));
+
+	return true;
+}
+
+LAYEREDLBSTRATEGY_TEST(GIVEN_layered_strat_of_two_by_four_grid_strat_singleton_strat_and_default_strat_WHEN_get_minimum_required_workers_called_THEN_ten_returned)
+{
+	AutomationOpenMap("/Engine/Maps/Entry");
+
+	TSharedPtr<TestData> Data = TSharedPtr<TestData>(new TestData);
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FCreateStrategy(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetDefaultLayer(Data, UGridBasedLBStrategy::StaticClass()));
+	ADD_LATENT_AUTOMATION_COMMAND(FAddLayer(Data, { UTwoByFourLBGridStrategy::StaticClass(), {} }));
+	ADD_LATENT_AUTOMATION_COMMAND(FAddLayer(Data, { UGridBasedLBStrategy::StaticClass(), {} }));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(Data, {}));
+	ADD_LATENT_AUTOMATION_COMMAND(FCheckMinimumWorkers(Data, this, 10));
+
+	return true;
+}
+
+LAYEREDLBSTRATEGY_TEST(Given_layered_strat_of_2_single_cell_strats_and_default_strat_WHEN_set_virtual_worker_ids_called_with_2_ids_THEN_error_is_logged)
+{
+	AutomationOpenMap("/Engine/Maps/Entry");
+
+	TSharedPtr<TestData> Data = TSharedPtr<TestData>(new TestData);
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FCreateStrategy(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetDefaultLayer(Data, UGridBasedLBStrategy::StaticClass()));
+	ADD_LATENT_AUTOMATION_COMMAND(FAddLayer(Data, { UGridBasedLBStrategy::StaticClass(), {ALayer1Pawn::StaticClass()} }));
+	ADD_LATENT_AUTOMATION_COMMAND(FAddLayer(Data, { UGridBasedLBStrategy::StaticClass(), {ALayer2Pawn::StaticClass()} }));
+	this->AddExpectedError("LayeredLBStrategy was not given enough VirtualWorkerIds to meet the demands of the layer strategies.",
+		EAutomationExpectedErrorFlags::MatchType::Contains, 1);
```

Comment:
I think might be worth moving this statement to either above or below all the `ADD_LATENT_AUTOMATION_COMMAND` functions for organisation sake. I think it's removes any ambiguity about the importance of execution order since all of ADD_LATENT_AUTOMATION_COMMAND aren't ran synchronously with where they appear here.

Reply:
yea I think where u have it in `Given_layered_strat_WHEN_set_local_worker_called_twice_THEN_an_error_is_logged` looks  good

Reply:
moved